src/services/db/GitFileSystemService.ts

```diff
@@ -337,6 +338,86 @@ export default class GitFileSystemService {
     )
   }
 
+  /**
+   * Wrapper over `git diff --name-only` that also creates a local tracking branch for `base` and/or `head` if they do not exist.
+   * Note that `base` and `head` should be branch names, not commit hashes or other usually valid arguments for `git diff`.
+   */
+  getFilesChanged(
+    repoName: string,
+    base = "master",
+    head = "staging"
+  ): ResultAsync<string[], GitFileSystemError> {
+    if (base === head) {
+      return okAsync([])
+    }
+    return this.createLocalTrackingBranchIfNotExists(repoName, base)
+      .andThen(() => this.createLocalTrackingBranchIfNotExists(repoName, head))
+      .andThen(() => {
+        const efsVolPath = this.getEfsVolPathFromBranch(head)
+        return ResultAsync.fromPromise(
+          this.git
+            .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+            .diff([`${base}..${head}`, "--name-only"]),
+          (error) => {
+            logger.error(
+              `Error when getting diff files between "${base}" and "${head}": ${error}, when trying to access ${efsVolPath}/${repoName}`
+            )
+
+            if (error instanceof GitError) {
+              return new GitFileSystemError(
+                "Unable to retrieve git diff info from disk"
+              )
+            }
+
+            return new GitFileSystemError("An unknown error occurred")
+          }
+        ).map((files) =>
+          files
+            .trim()
+            .split("\n")
+            .filter((file) => file !== "")
+        )
+      })
+  }
+
+  /**
+   * Get latest commit for a file path on a branch (including deleted files)
+   */
+  getLatestCommitOfPath(
+    repoName: string,
+    path: string,
```

Comment:
where is the path coming from? Is any part of this coming from a user input?

for e.g. if the path was "../../../<some other repo>/<path>" it should not be giving other repo's info

Reply:
if you're referring to user input via an endpoint then no it is not
the filenames from `getFilesChanged` are passed into this method
so it is not possible to receive a malicious path here
furthermore the git log is only done in their repo as seen by `git.cwd({path: efsvolPath.....`